File: tags/IVEF_0_1_0/ivef-objc/Classes/IVEFVoyage.m

#import "IVEFVoyage.h"


@implementation IVEFVoyage

- (id) init {
    self = [super init];
    if (self != nil) {
        m_cargoTypePresent = false;
        m_destinationPresent = false;
        m_ETAPresent = false;
        m_ATAPresent = false;
        m_airDraughtPresent = false;
        m_draughtPresent = false;
        m_dateFormatter = [[NSDateFormatter alloc] init];
        [m_dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS"];
    }
    return self;
}

-(void) setId:(NSString *) val {

    [m_id release];
    m_id = val;
    [m_id retain];
}

- (NSString *) getId {

    return m_id;
}

-(void) setSourceName:(NSString *) val {

    [m_sourceName release];
    m_sourceName = val;
    [m_sourceName retain];
}

- (NSString *) getSourceName {

    return m_sourceName;
}

-(void) setSource:(int) val {

    if ( ( val != 1 ) &&
         ( val != 2 ) &&
         ( val != 3 ) )
        return;
    m_source = val;
}

- (int) getSource {

    return m_source;
}

-(void) setCargoType:(int) val {

    if ( ( val != 0 ) &&
         ( val != 1 ) &&
         ( val != 2 ) &&
         ( val != 3 ) &&
         ( val != 4 ) &&
         ( val != 9 ) )
        return;
    m_cargoTypePresent = true;
    m_cargoType = val;
}

- (int) getCargoType {

    return m_cargoType;
}

-(bool) hasCargoType {

    return m_cargoTypePresent;
}

-(void) setDestination:(NSString *) val {

    m_destinationPresent = true;
    [m_destination release];
    m_destination = val;
    [m_destination retain];
}

- (NSString *) getDestination {

    return m_destination;
}

-(bool) hasDestination {

    return m_destinationPresent;
}

-(void) setETA:(NSDate *) val {

    m_ETAPresent = true;
    [m_ETA release];
    m_ETA = val;
    [m_ETA retain];
}

- (NSDate *) getETA {

    return m_ETA;
}

-(bool) hasETA {

    return m_ETAPresent;
}

-(void) setATA:(NSDate *) val {

    m_ATAPresent = true;
    [m_ATA release];
    m_ATA = val;
    [m_ATA retain];
}

- (NSDate *) getATA {

    return m_ATA;
}

-(bool) hasATA {

    return m_ATAPresent;
}

-(void) setAirDraught:(float) val {

    m_airDraughtPresent = true;
    m_airDraught = val;
}

- (float) getAirDraught {

    return m_airDraught;
}

-(bool) hasAirDraught {

    return m_airDraughtPresent;
}

-(void) setDraught:(float) val {

    m_draughtPresent = true;
    m_draught = val;
}

- (float) getDraught {

    return m_draught;
}

-(bool) hasDraught {

    return m_draughtPresent;
}

-(void) setAttributes:(NSDictionary *)attributeDict {

        for (NSString *key in attributeDict) {
            if ([key isEqualToString: @"Id"]) {
                NSString *val = [attributeDict objectForKey: key];
                [self setId: val];
            }
            else if ([key isEqualToString:@"SourceName"]) {
                NSString *val = [attributeDict objectForKey: key];
                [self setSourceName: val];
            }
            else if ([key isEqualToString:@"Source"]) {
                NSString *value = [attributeDict objectForKey: key];
                int val = [value intValue];
                [self setSource: val];
            }
            else if ([key isEqualToString:@"CargoType"]) {
                NSString *value = [attributeDict objectForKey: key];
                int val = [value intValue];
                [self setCargoType: val];
            }
            else if ([key isEqualToString:@"Destination"]) {
                NSString *val = [attributeDict objectForKey: key];
                [self setDestination: val];
            }
            else if ([key isEqualToString:@"ETA"]) {
                NSString *value = [attributeDict objectForKey: key];
                NSDate *val = [m_dateFormatter dateFromString: value]; // assume "yyyy-MM-ddThh:mm:ss.zzz"
                [self setETA: val];
            }
            else if ([key isEqualToString:@"ATA"]) {
                NSString *value = [attributeDict objectForKey: key];
                NSDate *val = [m_dateFormatter dateFromString: value]; // assume "yyyy-MM-ddThh:mm:ss.zzz"
                [self setATA: val];
            }
            else if ([key isEqualToString:@"AirDraught"]) {
                NSString *value = [attributeDict objectForKey: key];
                float val = [value floatValue];
                [self setAirDraught: val];
            }
            else if ([key isEqualToString:@"Draught"]) {
                NSString *value = [attributeDict objectForKey: key];
                float val = [value floatValue];
                [self setDraught: val];
            }
        }
}

-(NSString *) XML {

    NSMutableString *xml = [NSMutableString stringWithString:@"<Voyage"];
    [xml appendString: @" Id=\""];
    [xml appendString: m_id];
    [xml appendString: @"\""];
    [xml appendString: @" SourceName=\""];
    [xml appendString: m_sourceName];
    [xml appendString: @"\""];
    [xml appendString: @" Source=\""];
    [xml appendString: [NSString stringWithFormat:@"%d", m_source]];
    [xml appendString: @"\""];
    if ( [self hasCargoType] ) {
        [xml appendString: @" CargoType=\""];
        [xml appendString: [NSString stringWithFormat:@"%d", m_cargoType]];
        [xml appendString: @"\""];
    }
    if ( [self hasDestination] ) {
        [xml appendString: @" Destination=\""];
        [xml appendString: m_destination];
        [xml appendString: @"\""];
    }
    if ( [self hasETA] ) {
        [xml appendString: @" ETA=\""];
        [xml appendString: [m_ETA descriptionWithCalendarFormat:@"%Y-%m-%dT%H:%M:%S.%F" timeZone:nil locale:[[NSUserDefaults standardUserDefaults] dictionaryRepresentation]]];
        [xml appendString: @"\""];
    }
    if ( [self hasATA] ) {
        [xml appendString: @" ATA=\""];
        [xml appendString: [m_ATA descriptionWithCalendarFormat:@"%Y-%m-%dT%H:%M:%S.%F" timeZone:nil locale:[[NSUserDefaults standardUserDefaults] dictionaryRepresentation]]];
        [xml appendString: @"\""];
    }
    if ( [self hasAirDraught] ) {
        [xml appendString: @" AirDraught=\""];
        [xml appendString: [NSString stringWithFormat:@"%f", m_airDraught]];
        [xml appendString: @"\""];
    }
    if ( [self hasDraught] ) {
        [xml appendString: @" Draught=\""];
        [xml appendString: [NSString stringWithFormat:@"%f", m_draught]];
        [xml appendString: @"\""];
    }
    [xml appendString:@">\n"];
    [xml appendString: @"</Voyage>\n"];
    return xml;
}

-(NSString *) stringValueWithLead: (NSString *) lead {

    NSMutableString *str = [[[NSMutableString alloc] init] autorelease];
    [str setString: [lead stringByAppendingString:@"Voyage\n"]];
    [str appendString: [lead stringByAppendingString: @" "]];
    [str appendString: @"Id=\""];
    [str appendString: m_id];
    [str appendString: @"\"\n"];

    [str appendString: [lead stringByAppendingString: @" "]];
    [str appendString: @"SourceName=\""];
    [str appendString: m_sourceName];
    [str appendString: @"\"\n"];

    [str appendString: [lead stringByAppendingString: @" "]];
    [str appendString: @"Source=\""];
    [str appendString: [NSString stringWithFormat:@"%d", m_source]];
    [str appendString: @"\"\n"];

    if ( [self hasCargoType] ) {
        [str appendString: [lead stringByAppendingString: @" "]];
        [str appendString: @"CargoType = \""];
        [str appendString: [NSString stringWithFormat:@"%d", m_cargoType]];
    }
    if ( [self hasDestination] ) {
        [str appendString: [lead stringByAppendingString: @" "]];
        [str appendString: @"Destination = \""];
        [str appendString: m_destination];
    }
    if ( [self hasETA] ) {
        [str appendString: [lead stringByAppendingString: @" "]];
        [str appendString: @"ETA = \""];
        [str appendString: [m_ETA descriptionWithCalendarFormat:@"%Y-%m-%dT%H:%M:%S.%F" timeZone:nil locale:[[NSUserDefaults standardUserDefaults] dictionaryRepresentation]]];
    }
    if ( [self hasATA] ) {
        [str appendString: [lead stringByAppendingString: @" "]];
        [str appendString: @"ATA = \""];
        [str appendString: [m_ATA descriptionWithCalendarFormat:@"%Y-%m-%dT%H:%M:%S.%F" timeZone:nil locale:[[NSUserDefaults standardUserDefaults] dictionaryRepresentation]]];
    }
    if ( [self hasAirDraught] ) {
        [str appendString: [lead stringByAppendingString: @" "]];
        [str appendString: @"AirDraught = \""];
        [str appendString: [NSString stringWithFormat:@"%f", m_airDraught]];
    }
    if ( [self hasDraught] ) {
        [str appendString: [lead stringByAppendingString: @" "]];
        [str appendString: @"Draught = \""];
        [str appendString: [NSString stringWithFormat:@"%f", m_draught]];
    }
    return str;
}


@end

File: ivef-lib/tags/IVEF_0_1_2/ivef-qt/IVEFVoyage.cpp

#include "IVEFVoyage.h"

Voyage::Voyage() {

    m_id = "";
    m_sourceName = "";
    m_source = 0;
    m_cargoType = 0;
    m_cargoTypePresent = false;
    m_destination = "";
    m_destinationPresent = false;
    m_ETA = QDateTime();
    m_ETAPresent = false;
    m_ATA = QDateTime();
    m_ATAPresent = false;
    m_personsOnBoard = 0.0;
    m_personsOnBoardPresent = false;
    m_airDraught = 0.0;
    m_airDraughtPresent = false;
    m_draught = 0.0;
    m_draughtPresent = false;
}

Voyage::Voyage(const Voyage &val) : QObject() {

    m_id = val.m_id;
    m_sourceName = val.m_sourceName;
    m_source = val.m_source;
    m_cargoTypePresent = val.m_cargoTypePresent;
    m_cargoType = val.m_cargoType;
    m_destinationPresent = val.m_destinationPresent;
    m_destination = val.m_destination;
    m_ETAPresent = val.m_ETAPresent;
    m_ETA = val.m_ETA;
    m_ATAPresent = val.m_ATAPresent;
    m_ATA = val.m_ATA;
    m_personsOnBoardPresent = val.m_personsOnBoardPresent;
    m_personsOnBoard = val.m_personsOnBoard;
    m_airDraughtPresent = val.m_airDraughtPresent;
    m_airDraught = val.m_airDraught;
    m_draughtPresent = val.m_draughtPresent;
    m_draught = val.m_draught;
}

Voyage & Voyage::operator=(const Voyage &val) {

    m_id = val.m_id;
    m_sourceName = val.m_sourceName;
    m_source = val.m_source;
    m_cargoTypePresent = val.m_cargoTypePresent;
    m_cargoType = val.m_cargoType;
    m_destinationPresent = val.m_destinationPresent;
    m_destination = val.m_destination;
    m_ETAPresent = val.m_ETAPresent;
    m_ETA = val.m_ETA;
    m_ATAPresent = val.m_ATAPresent;
    m_ATA = val.m_ATA;
    m_personsOnBoardPresent = val.m_personsOnBoardPresent;
    m_personsOnBoard = val.m_personsOnBoard;
    m_airDraughtPresent = val.m_airDraughtPresent;
    m_airDraught = val.m_airDraught;
    m_draughtPresent = val.m_draughtPresent;
    m_draught = val.m_draught;
    return *this;
}

QString Voyage::encode( QString str) {

    str.replace('&', "&amp;");
    str.replace('<', "&lt;");
    str.replace('>', "&gt;");
    str.replace('"', "&quot;");
    return str;
}

void Voyage::setId(QString val) {

    m_id = val;
}

QString Voyage::getId() const {

    return m_id;
}

void Voyage::setSourceName(QString val) {

    m_sourceName = val;
}

QString Voyage::getSourceName() const {

    return m_sourceName;
}

void Voyage::setSource(int val) {

    if ( ( val != 1 ) &&
         ( val != 2 ) &&
         ( val != 3 ) )
        return;
    m_source = val;
}

int Voyage::getSource() const {

    return m_source;
}

void Voyage::setCargoType(int val) {

    if ( ( val != 0 ) &&
         ( val != 1 ) &&
         ( val != 2 ) &&
         ( val != 3 ) &&
         ( val != 4 ) &&
         ( val != 9 ) )
        return;
    m_cargoTypePresent = true;
    m_cargoType = val;
}

int Voyage::getCargoType() const {

    return m_cargoType;
}

bool Voyage::hasCargoType() {

    return m_cargoTypePresent;
}

void Voyage::setDestination(QString val) {

    m_destinationPresent = true;
    m_destination = val;
}

QString Voyage::getDestination() const {

    return m_destination;
}

bool Voyage::hasDestination() {

    return m_destinationPresent;
}

void Voyage::setETA(QDateTime val) {

    m_ETAPresent = true;
    m_ETA = val;
}

QDateTime Voyage::getETA() const {

    return m_ETA;
}

bool Voyage::hasETA() {

    return m_ETAPresent;
}

void Voyage::setATA(QDateTime val) {

    m_ATAPresent = true;
    m_ATA = val;
}

QDateTime Voyage::getATA() const {

    return m_ATA;
}

bool Voyage::hasATA() {

    return m_ATAPresent;
}

void Voyage::setPersonsOnBoard(float val) {

    m_personsOnBoardPresent = true;
    m_personsOnBoard = val;
}

float Voyage::getPersonsOnBoard() const {

    return m_personsOnBoard;
}

bool Voyage::hasPersonsOnBoard() {

    return m_personsOnBoardPresent;
}

void Voyage::setAirDraught(float val) {

    m_airDraughtPresent = true;
    m_airDraught = val;
}

float Voyage::getAirDraught() const {

    return m_airDraught;
}

bool Voyage::hasAirDraught() {

    return m_airDraughtPresent;
}

void Voyage::setDraught(float val) {

    m_draughtPresent = true;
    m_draught = val;
}

float Voyage::getDraught() const {

    return m_draught;
}

bool Voyage::hasDraught() {

    return m_draughtPresent;
}

QString Voyage::toXML() {

    QString xml = "<Voyage";
    xml.append(" Id=\"" + encode (m_id) + "\"");
    xml.append(" SourceName=\"" + encode (m_sourceName) + "\"");
    xml.append(" Source=\"" + QString::number(m_source) + "\"");
    if ( hasCargoType() ) {
        xml.append(" CargoType=\"" + QString::number(m_cargoType) + "\"");
    }
    if ( hasDestination() ) {
        xml.append(" Destination=\"" + encode (m_destination) + "\"");
    }
    if ( hasETA() ) {
        xml.append(" ETA=\"" + m_ETA.toString("yyyy-MM-ddThh:mm:ss.zzzZ") + "\"");
    }
    if ( hasATA() ) {
        xml.append(" ATA=\"" + m_ATA.toString("yyyy-MM-ddThh:mm:ss.zzzZ") + "\"");
    }
    if ( hasPersonsOnBoard() ) {
        xml.append(" PersonsOnBoard=\"" + QString::number(m_personsOnBoard) + "\"");
    }
    if ( hasAirDraught() ) {
        xml.append(" AirDraught=\"" + QString::number(m_airDraught) + "\"");
    }
    if ( hasDraught() ) {
        xml.append(" Draught=\"" + QString::number(m_draught) + "\"");
    }
    xml.append(">\n");
    xml.append( "</Voyage>\n");
    return xml;
}

QString Voyage::toString(QString lead) {

    QString str = lead + "Voyage\n";
    str.append( lead + "    Id = " + m_id + "\n");
    str.append( lead + "    SourceName = " + m_sourceName + "\n");
    str.append( lead + "    Source = " + QString::number(m_source) + "\n");
    if ( hasCargoType() ) {
        str.append( lead + "    CargoType = " + QString::number(m_cargoType) + "\n");
    }
    if ( hasDestination() ) {
        str.append( lead + "    Destination = " + m_destination + "\n");
    }
    if ( hasETA() ) {
        str.append( lead + "    ETA = " + m_ETA.toString("yyyy-MM-ddThh:mm:ss.zzzZ") + "\n");
    }
    if ( hasATA() ) {
        str.append( lead + "    ATA = " + m_ATA.toString("yyyy-MM-ddThh:mm:ss.zzzZ") + "\n");
    }
    if ( hasPersonsOnBoard() ) {
        str.append( lead + "    PersonsOnBoard = " + QString::number(m_personsOnBoard) + "\n");
    }
    if ( hasAirDraught() ) {
        str.append( lead + "    AirDraught = " + QString::number(m_airDraught) + "\n");
    }
    if ( hasDraught() ) {
        str.append( lead + "    Draught = " + QString::number(m_draught) + "\n");
    }
    return str;
}

